modules/apps/frontend-js/frontend-js-spa-web/src/main/java/com/liferay/frontend/js/spa/web/internal/servlet/taglib/SPATopHeadJSPDynamicInclude.java

```diff
@@ -101,19 +101,16 @@ public void include(
 			"validStatusCodes", _spaHelper.getValidStatusCodesJSONArray()
 		);
 
-		String initModuleName = _npmResolver.resolveModuleName(
-			"frontend-js-spa-web/init");
+		StringBundler javascriptSB = new StringBundler();
 
-		ScriptData initScriptData = new ScriptData();
+		javascriptSB.append("frontendJsSpaWebInit.default(");
+		javascriptSB.append(configJSONObject.toJSONString());
+		javascriptSB.append(");");
 
-		initScriptData.append(
-			null,
-			"frontendJsSpaWebInit.default(" + configJSONObject.toJSONString() +
-				")",
-			initModuleName + " as frontendJsSpaWebInit",
-			ScriptData.ModulesType.ES6);
-
-		initScriptData.writeTo(httpServletResponse.getWriter());
+		_jsLoaderSupport.writeScript(
+			httpServletResponse.getWriter(), "frontend-js-spa-web",
+			"frontendJsSpaWebInit", javascriptSB.toString(),
+			"window.setTimeout(function() {[$JAVASCRIPT_CODE$]}, 0);");
```

Comment:
This timeout is necessary to delay the execution of the `getModule` call until the page has been loaded. This is because `getModule` appends to `document.body` and, because this is the SPA module, there's no `body` yet.

Other options are:

1. Use `window.onload`
2. Make `getModule` write to `head` instead of body (though I would have to check this because I think I used  `body` instead of `head` for a reason).

Reply:
I have some doubts regarding this now. Not 100% sure that the timeout is going to synchronize with the HTML rendering engine... :thinking: 

Opinions?

I think I'll change it to `onload` or change `getModule` just to be safe...

Reply:
I've changed this. See [this commit](https://github.com/izaera/liferay-portal/commit/8d1b42b894bf014d11a47e5bbaa768e93bf24541).

Now we don't need setTimeout or onload any more.